jme3-core/src/main/java/com/jme3/scene/control/LightControl.java

```diff
@@ -127,49 +130,58 @@ protected void controlUpdate(float tpf) {
         }
     }
 
+    /**
+     * Sets the light to adopt the spatial's world transformations.
+     *
+     * @author Markil 3
+     * @author pspeed42
+     */
     private void spatialToLight(Light light) {
-
         final Vector3f worldTranslation = spatial.getWorldTranslation();
+        final Vector3f worldDirection = spatial.getWorldRotation().mult(Vector3f.UNIT_Z).negateLocal();
```

Comment:
The new `spatialToLight()` method generated a garbage `Vector3f` each time it is invoked. Please use `TempVars` to avoid this.

Reply:
Done.